File: sources/scala/08_average.scala
// task 08 average — expected output: 0.498046875
// build: scalac -release 17 08_average.scala    run: scala Main

object Main {
  def main(args: Array[String]): Unit = {
    var total = 0.0
    var i = 0L
    while (i < 100000000L) {
      val reading = (i % 256L).toDouble / 256.0
      total += reading
      i += 1L
    }
    println(total / 100000000.0)
  }
}

File: sources/nim/_08_average.nim
# task 08 average — expected output: 0.498046875
# build: nim c -d:release -o:prog _08_average.nim    run: ./prog

var total = 0.0
for i in 0 ..< 100_000_000:
  let reading = float(i mod 256) / 256.0
  total += reading
let average = total / 100_000_000.0
echo average
